TACC/Core-Portal
# Task/wp 209 fix deprecated warnings (part II)

## Overview
This is a continuation from [WP-171](https://jira.tacc.utexas.edu/browse/WP-171). 
When running unit tests, a lot of deprecated warnings were appearing. This tickets addresses those warnings. 

## Related
* [WP-209](https://jira.tacc.utexas.edu/browse/WP-209)

## Changes
- The previous ticket [WP-171] addressed issues with our code (Core-Portal). However, a lot of the warnings being seen are issues with Django. To keep these from showing up, they are filtered out; `pytest.ini` now has `'filterwanrings'`. 

- `PytestUnknownMarkWarning: Unknown pytest.mark.asyncio` was addressed by adding `markers = asyncio: mark a test as asyncio` to `pytest.ini`

- `RemovedInDjango40Warning: The providing_args argument is deprecated.` was addressed in `signals.py` by redefining `portal_event` as suggested in the Django documentation.


## Testing
1. Start docker container —> `docker exec -it core_portal_django bash`
2. Run `pytest -ra portal/apps/auth/api/`
3. Notice there are no more deprecated warnings.

## UI
![Screen Shot 2023-08-24 at 12 32 07 PM](https://github.com/TACC/Core-Portal/assets/35277477/d8db0bad-b043-4b66-802a-2af0c7bdbfd7)

## Notes
Running `pytest -ra portal/apps/` shows that some tests are failing, and also has 7 warnings (not deprecation warnings). These might need to be addressed.
